draft-ietf-taps-impl.md

```diff
@@ -1091,7 +1091,7 @@ Receive:
 : RECEIVE.SCTP. The "partial flag" of RECEIVE.SCTP invokes a `ReceivedPartial` event.
 
 Close:
-If this is the only Connection object that is assigned to the SCTP association, CLOSE.SCTP is called. Else, the Connection object is one out of several Connection objects that are assigned to the same SCTP assocation, and RESET_STREAM.SCTP must be called, which informs the peer that the stream will no longer be used for mapping and can be used by future `Initiate`, `InitiateWithSend` or `Listen` calls. At the peer, the event RESET_STREAM-EVENT.SCTP will fire, which the peer must answer by issuing RESET_STREAM.SCTP too. The resulting local RESET_STREAM-EVENT.SCTP informs the transport system that the stream number can now be re-used by the next `Initiate`, `InitiateWithSend` or `Listen` calls.
+If this is the only Connection object that is assigned to the SCTP association, CLOSE.SCTP is called, and the `Closed` event is immediately delivered to the application (because SCTP does not support half-closed connections). Else, the Connection object is one out of several Connection objects that are assigned to the same SCTP assocation, and RESET_STREAM.SCTP must be called, which informs the peer that the stream will no longer be used for mapping and can be used by future `Initiate`, `InitiateWithSend` or `Listen` calls. At the peer, the event RESET_STREAM-EVENT.SCTP will fire, which the peer must answer by issuing RESET_STREAM.SCTP too. The resulting local RESET_STREAM-EVENT.SCTP informs the transport system that the stream number can now be re-used by the next `Initiate`, `InitiateWithSend` or `Listen` calls.
 
```

Comment:
"the `Closed` event is immediately delivered to the application" may sound a bit misleading as data sent from either end but not yet delivered is reliably delivered before this event fires I think. So you can still receive data after issuing the Close if it was already in the peers send queue, right?

Do we not need to also specify when the 'Closed' event should fire in the multistreamed case. After getting the RESET_STREAM-EVENT.SCTP back I guess?

Reply:
1) Right! in the peers send queue, or in flight, or in the local host's receive buffer, I believe. Would it be ok to fix this as follows:
If this is the only Connection object that is assigned to the SCTP association, CLOSE.SCTP `Closed` event is delivered to the application when this call returns.
[no need for us to go into these specifics here? Close semantics are given in the interface document anyway.]

2) Right! Would it be ok to fix this as follows:
"The resulting local RESET_STREAM-EVENT.SCTP informs the transport system that the stream number can now be re-used by the next `Initiate`, `InitiateWithSend` or `Listen` calls, and invokes a `Closed` event towards the application.
[I suppose that "after delivering any data from the local receive buffer" could be added, but I also don't think that this adds clarity, and so I would just formulate it as above).

Reply:
For 1. perhaps "... when the call is completed". I guess the call will return immediately, but the close event is delivered to the application when the shutdown complete event is received.

2. looks good!

Reply:
About 1) also true, thanks - silly me, it should be upon "CLOSE-EVENT.SCTP" from RFC 8303. I once wrote that, after all  :-)